src/sst.rs

```diff
@@ -393,34 +383,47 @@ impl<'a> EncodedSsTableBuilder<'a> {
         }
     }
 
-    pub fn add(
-        &mut self,
-        key: &[u8],
-        value: Option<&[u8]>,
-        attrs: RowAttributes,
-    ) -> Result<(), SlateDBError> {
+    pub fn add(&mut self, entry: RowEntry) -> Result<(), SlateDBError> {
         self.num_keys += 1;
+        let key = entry.key.clone();
 
-        if !self.builder.add(key, value, attrs.clone()) {
+        if !self.builder.add(entry.clone()) {
             // Create a new block builder and append block data
             if let Some(block) = self.finish_block()? {
                 self.current_len += block.len();
                 self.blocks.push_back(Bytes::from(block));
             }
 
             // New block must always accept the first KV pair
-            assert!(self.builder.add(key, value, attrs));
-            self.first_key = Some(self.index_builder.create_vector(key));
+            assert!(self.builder.add(entry));
+            self.first_key = Some(self.index_builder.create_vector(&key));
         } else if self.sst_first_key.is_none() {
-            self.sst_first_key = Some(Bytes::copy_from_slice(key));
-            self.first_key = Some(self.index_builder.create_vector(key));
+            self.sst_first_key = Some(Bytes::copy_from_slice(&key));
+            self.first_key = Some(self.index_builder.create_vector(&key));
         }
 
-        self.filter_builder.add_key(key);
+        self.filter_builder.add_key(&key);
 
         Ok(())
     }
 
+    #[cfg(test)]
+    pub fn add_kv(
```

Comment:
Seeing this in a few places. Maybe put it in `test_utils.rs`?

Reply:
make sense, let me move it :+1:

Reply:
Sounds good!